checkstyle.xml

```diff
@@ -59,9 +59,6 @@
             <property name="tokens"
              value="LITERAL_TRY, LITERAL_FINALLY, LITERAL_IF, LITERAL_ELSE, LITERAL_SWITCH"/>
         </module>
-        <module name="NeedBraces">
-            <property name="severity" value="error"/>
-        </module>
```

Comment:
In my Opinion brackets should be required in c as well, because mixing styles is not a good Idea.

Reply:
I see a lot of typing coming our way then… instead of

    if (foo)
        do_stuff();

I now have to type

    if (foo) {
        do_stuff();
    }

Which means more lines for the same code, and less code that will fit on a page of screen (bad for legibility).

It’s not really mixing styles: braces around single-statement blocks are optional in both C and Java, but mandatory around multi-statement blocks (as there is no other way to tell where the block ends). So the style is either “always use braces” or “use braces where necessary”.

The only drawback of omitting braces around single-block statements is that diffs can become slightly cluttered when a single-statement block is later extended to multiple statements: not only is there a line for each added statement, but also one for the block statement (with the added opening brace) and another for the newly-added closing brace. But I think this is outweighed by the benefits of less typing and better legibility.

Reply:
okay, then i am okay with "use braces where necessary"